applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_cr_beam_element_3D2N.h

```diff
@@ -0,0 +1,77 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:     BSD License
+//           license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Martin Fusseder, https://github.com/MFusseder
+//
+
+
+#if !defined(ADJOINT_FINITE_DIFFERENCE_CR_BEAM_ELEMENT_H_INCLUDED )
+#define  ADJOINT_FINITE_DIFFERENCE_CR_BEAM_ELEMENT_H_INCLUDED
+
+#include "adjoint_finite_difference_base_element.h"
+
+namespace Kratos
+{
+
+class AdjointFiniteDifferenceCrBeamElement : public AdjointFiniteDifferencingBaseElement
+{
+public:
+    KRATOS_CLASS_POINTER_DEFINITION(AdjointFiniteDifferenceCrBeamElement);
+
+    typedef Element BaseType;
+    typedef BaseType::GeometryType GeometryType;
```

Comment:
do you need all of these?
It might be better to use `auto` in some places

Reply:
Done